caffe2/operators/hip/conv_op_miopen.hip

```diff
@@ -252,48 +253,101 @@ bool MIOPENConvOp::DoRunWithType() {
     VLOG(1) << "Changing MIOpen descriptor configurations.";
     if (input_changed) {
       mio_input_dims_ = X.sizes().vec();
-      MIOPEN_ENFORCE(miopenSet4dTensorDescriptor(
-          bottom_desc_, miopenTypeWrapper<T_X>::type, N, C, H, W));
+      if (mio_input_dims_.size() == 4) {
+        MIOPEN_ENFORCE(miopenSet4dTensorDescriptor(
+            bottom_desc_, miopenTypeWrapper<T_X>::type, N, C, H, W));
+      } else {
+        vector<int> input_dims (begin(mio_input_dims_), end(mio_input_dims_));
+        MIOPEN_ENFORCE(miopenSetTensorDescriptor(
+            bottom_desc_, miopenTypeWrapper<T_X>::type, input_dims.size(), input_dims.data(), nullptr));
+      }
     }
 
     if (weight_changed) {
       mio_weight_dims_ = Weight.sizes().vec();
-      MIOPEN_ENFORCE(miopenInitConvolutionDescriptor(
-          conv_desc_,
-          mode_,
-          pad_t(),
-          pad_l(),
-          stride_h(),
-          stride_w(),
-          dilation_h(),
-          dilation_w()));
+      if (mio_weight_dims_.size() == 4) {
+        MIOPEN_ENFORCE(miopenInitConvolutionDescriptor(
+            conv_desc_,
+            mode_,
+            pad_t(),
+            pad_l(),
+            stride_h(),
+            stride_w(),
+            dilation_h(),
+            dilation_w()));
+      } else {
+        int spatial_dim = 3;
+        MIOPEN_ENFORCE(miopenInitConvolutionNdDescriptor(
+            conv_desc_,
+            spatial_dim,
+            pads_.data(),
+            stride_.data(),
+            dilation_.data(),
+            mode_));
+      }
 
       MIOPEN_ENFORCE(miopenSetConvolutionGroupCount(conv_desc_, group_));
 
-      MIOPEN_ENFORCE(miopenSet4dTensorDescriptor(
-          weight_desc_,
-          miopenTypeWrapper<T_W>::type,
-          M,
-          C / group_,
-          kernel_h(),
-          kernel_w()));
+      if (mio_weight_dims_.size() == 4) {
+        MIOPEN_ENFORCE(miopenSet4dTensorDescriptor(
+            weight_desc_,
+            miopenTypeWrapper<T_W>::type,
+            M,
+            C / group_,
+            kernel_h(),
+            kernel_w()));
+      } else {
+        vector<int> weight_dims (begin(mio_weight_dims_), end(mio_weight_dims_));
+        MIOPEN_ENFORCE(miopenSetTensorDescriptor(
+            weight_desc_,
+            miopenTypeWrapper<T_W>::type,
+            mio_weight_dims_.size(),
+            weight_dims.data(),
+            nullptr));
+      }
     }
 
-    MIOPEN_ENFORCE(miopenGetConvolutionForwardOutputDim(
-        conv_desc_,
-        bottom_desc_,
-        weight_desc_,
-        &N_out,
-        &C_out,
-        &H_out,
-        &W_out));
+    
+    mio_output_dims_ = Y->sizes().vec();
+    int output_dim_size = mio_input_dims_.size(); 
+    vector<int> output_dims (begin(mio_output_dims_), end(mio_output_dims_));
```

Comment:
is this required to be done on every invocation?

Reply:
No, it is not done on every invocation. 
This is done only when the if (input_changed ) is true in Line number 252 and 470.